server/routes/controllers/usersController.js

```diff
@@ -92,37 +92,18 @@ const usersController = {
   },
 
   changePassword (req, res) {
-    const users = this.Users
     const {email, password} = req.body
 
-    users.update({'email': email},
-      {$set: {'password': password}}, (error, r) => {
-        if (error) {
-          console.error(error)
-          return res.sendStatus(500)
-        }
-        if (r) {
-          if (r.n === 0) {
-            console.error(`Email doesn't exist`)
-            res.statusMessage = `Email doesn't exist`
-            return res.sendStatus(500)
-          } else {
-            if (r.nModified === 1) {
-              console.log(`New password for ${email} saved successfully!`)
-            }
-            users.findOne({ email }, (err, user) => {
-              if (err) {
-                console.error(err)
-                return res.sendStatus(500)
-              }
-
-              if (user) {
-                return res.send(user.toJSON())
-              }
-            })
-          }
-        }
-      })
+    var query = {'email': email}
+    this.Users.findOneAndUpdate(query, {$set: { password: password }}, {}, function (err, doc) {
+      if (err) {
```

Comment:
Great improvement in the logic!!

Might even go one step further (note that this assumes you changed `err` to `error`):

```
if (err) {
  return res.send(500, { error })
else if (!doc) {
  return res.send(404, { error: 'No user found. No password updated' })
}

return res.send(doc.toJSON())
```

This has the added bonus of separating your error cases into one conditional block, distinct from your happy path.

Reply:
Thanks. I'll do it